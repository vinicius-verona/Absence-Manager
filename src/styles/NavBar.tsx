import styled from "styled-components";

export const Nav = styled.nav`
    background-color: ${({ theme }) => theme.palettes.primary};
    padding: 0px 30px;
    min-width: 500px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    position: sticky;
    z-index: 10;
    top: 0;
    left: 0;

    @media screen and (max-width: 500px) {
        padding: 0px 10%;
        justify-content: center;
    }
`;

export const NavTitle = styled.div`
    width: 100vw;
    height: 100%;
    align-items: center;
    position: relative;
`;

export const Title = styled.h1`
    color: white;
    font-weight: 600;
    text-align: center;
    font-size: 1.5rem;
`;

export const Logo = styled.img`
    height: 1.125rem;
    position: absolute;

    @media screen and (max-width: 690px) {
        visibility: hidden;
    }
`;
