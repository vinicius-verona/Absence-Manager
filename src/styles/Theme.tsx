// eslint-disable-next-line import/no-anonymous-default-export
export default {
    palettes: {
        primary: "rgba(255, 148, 25, 1)",
        lightPrimary: "rgba(255, 148, 25, 0.5)",
        light: "#DFDFDF",
        darkerLight: "#B5B5B5",
        bg_dark: "rgba(0, 0, 0, 0.5)",
    },
};
