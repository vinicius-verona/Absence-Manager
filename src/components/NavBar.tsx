import { Nav, NavTitle, Title, Logo } from "../styles/NavBar";

export default function NavBar() {
  return (
    <Nav>
      <Logo
        src={
          "https://d33wubrfki0l68.cloudfront.net/f140ee3478696725fd905f311242bfc48579b2f5/dcaca/images/logo_crewmeister_white.svg"
        }
        alt={"Crewmeister's logo"}
        className={"logo"}
      />
      <NavTitle>
        <Title>Absence Manager</Title>
      </NavTitle>
    </Nav>
  );
}
