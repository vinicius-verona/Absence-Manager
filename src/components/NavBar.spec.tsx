import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Theme from "../styles/Theme";
import NavBar from "./NavBar";

describe("Render NavBar on page", () => {
  it("Should render logo", () => {
    const { getByAltText } = render(
      <ThemeProvider theme={Theme}>
        <NavBar />
      </ThemeProvider>
    );

    expect(getByAltText("Crewmeister's logo")).toBeInTheDocument();
  });

  it("Should render title", () => {
    const { getByText } = render(
      <ThemeProvider theme={Theme}>
        <NavBar />
      </ThemeProvider>
    );

    expect(getByText("Absence Manager")).toBeInTheDocument();
  });
});
