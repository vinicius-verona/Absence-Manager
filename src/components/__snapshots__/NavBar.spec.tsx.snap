// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar on page Should match navbar snapshot 1`] = `
<nav
  className="sc-bdvvtL hyHtAJ"
>
  <img
    alt="Crewmeister's logo"
    className="sc-hKwDye jXmRyY logo"
    src="https://d33wubrfki0l68.cloudfront.net/f140ee3478696725fd905f311242bfc48579b2f5/dcaca/images/logo_crewmeister_white.svg"
  />
  <div
    className="sc-gsDKAQ kHNdgz"
  >
    <h1
      className="sc-dkPtRN fwKpHR"
    >
      Absence Manager
    </h1>
  </div>
</nav>
`;
